core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestSimplifyExpressions.java

```diff
@@ -256,23 +290,28 @@ public void testCastDateToBoundedVarchar()
     }
 
     private static void assertSimplifies(@Language("SQL") String expression, @Language("SQL") String expected)
+    {
+        assertSimplifies(expression, expected, BOOLEAN);
```

Comment:
There are many usages in this file where the expected type is not `BOOLEAN`, so this would appear to be incorrect.

Reply:
- Renamed the default `assertSimplifies`(the expression type is `BOOLEAN`) to `assertSimplifiesWithBooleanType`.
- Add type to the each test case